### ISSUE TITLE
EclipseLink is executing updating queries whenever an entity with a Set property is read

### ISSUE THREAD

**Issue by FabriSr** [OpenLiberty/open-liberty#17876]
**Describe the bug**
EclipseLink is failing to track changes when an entity has an embedded collection property mapped using Set's: whenever an entity instance is read from the entity manager update queries are executed against the database despite no entity data has been changed. 

For example, an entity:

```
@Entity
@Table(name = "employee")
@NamedQuery(name = "Employee.findAll", query = "SELECT e FROM Employee e ORDER BY e.id")
public class Employee implements Serializable
{
    .....
    
    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(
            name = "employee_phone_number",            
            joinColumns = @JoinColumn(name = "employeeId")
    )
    @AttributeOverride(name = "number", column = @Column(name = "number"))
    private Set<PhoneNumber> phoneNumbers;
}
```

Whenever the EmployeeDAO.readAll method is executed, update queries are executed against database:

```
@Transactional
@ApplicationScoped
public class EmployeeDAO
{
    @PersistenceContext(name = "jpa-unit")
    private EntityManager entityManager;
   
    public List<Employee> readAll()
    {        
        return this.entityManager.createNamedQuery("Employee.findAll", Employee.class).getResultList();
    }        
}
```

**Steps to Reproduce**
I have attached a small project to demonstrate the error. The project has two mapped entities: Employee and Person. Employee is mapped using Set collection type. Person is mapped using List. Only Employee instances are updated. EmployeeDAO does not change the state of the objects in the readAll method. The "populate" method creates objects only when the table is empty.

1 - Download and unzip the attached demo project [error_demo.zip](https://github.com/OpenLiberty/open-liberty/files/6826739/error_demo.zip)
2 - Navigate into the project root folder and run: $ mvn liberty:dev
3 - Open your brownser: http://localhost:9080/
4 - Open the jpa log file (./target/liberty/wlp/usr/servers/defaultServer/logs/jpa.log) and search for UpdateObjectQuery
5 - Refresh the page (whenever the page is loaded the readAll operation is executed). New UpdateObjectQuery entries are added to log file (indicating the update operation).

**Expected behavior**
Update queries are only performed when entity data changes (just as when using List)

**Comment by dazey3** [comment]
Hello! Thank you for bringing this issue to our attention!

Have you noticed this issue recently in a previously functioning application (regression) or are you investigating a new application currently in development?

My first suggestion (purely for investigation) would be to remove the @AttributeOverride from the Employee @ElementCollection mapping and see if the issue still exists. I suggest this because this example seems very similar to an issue I fixed last year with EclipseLink + (@AttributeOverride on @ElementCollection):
https://bugs.eclipse.org/bugs/show_bug.cgi?id=564260

I will download and look at the provided example so that I can attempt to confirm the issue as well. Thanks!

**Comment by dazey3** [comment]
This issue will be addressed in the EclipseLink open source community with https://github.com/eclipse-ee4j/eclipselink/issues/1217

**Comment by FabriSr** [comment]
Issue closed. Addressed in [eclipse-ee4j/eclipselink#1217](https://github.com/eclipse-ee4j/eclipselink/issues/1217).
